cloudinit/util.py

```diff
@@ -1010,13 +1011,24 @@ def read_conf_d(confd):
     # Load them all so that they can be merged
     cfgs = []
     for fn in confs:
-        cfgs.append(read_conf(os.path.join(confd, fn)))
+        try:
+            cfgs.append(read_conf(os.path.join(confd, fn)))
+        except OSError as e:
+            if e.errno == EACCES:
+                LOG.warning(
+                    "REDACTED config part %s/%s for non-root user", confd, fn
+                )
 
     return mergemanydict(cfgs)
 
 
 def read_conf_with_confd(cfgfile):
-    cfg = read_conf(cfgfile)
+    try:
+        cfg = read_conf(cfgfile)
+    except OSError as e:
+        if e.errno == EACCES:
+            LOG.warning("REDACTED config part %s for non-root user", cfgfile)
+        raise
```

Comment:
I don't think we want a raise here, we want to proceed to other "conf_d" files if available.

Reply:
Reworked to proceed to other "conf_d files" if available.